### ISSUE TITLE
How to change the vertical distance to the ground of the AR label?

### ISSUE THREAD

**Issue by Lilyxxzjzljy** [appoly/ARCore-Location#127]
when using the ARCore-Location to put AR labels at specific gps location,I want to put the AR label on the ground.I try to change the y-component of the translation:
```
          float y = frame.getCamera().getDisplayOrientedPose().ty() -15.8f;  //here is my code
                //float y=0f;
                if (marker.anchorNode != null && marker.anchorNode.getAnchor() != null) {
                    marker.anchorNode.getAnchor().detach();
                    marker.anchorNode.setAnchor(null);
                    marker.anchorNode.setEnabled(false);
                    marker.anchorNode = null;
                }
                // Don't immediately assign newly created anchor in-case of exceptions
                Pose translation = Pose.makeTranslation(xRotated, y, zRotated);
                Anchor newAnchor = mSession.createAnchor(
                        frame.getCamera()
                                .getDisplayOrientedPose()
                                .compose(translation)
                                .extractTranslation()
                );
                marker.anchorNode = new LocationNode(newAnchor, marker, this);
                marker.anchorNode.setScalingMode(LocationMarker.ScalingMode.GRADUAL_FIXED_SIZE);
                marker.anchorNode.setParent(mArSceneView.getScene());
                marker.anchorNode.addChild(mLocationMarkers.get(i).node);
                marker.node.setLocalPosition(Vector3.zero());
```

in the LocationScene.refreshAnchorsIfRequired,In my opinion,the unit of translation is meter,frame.getCamera().getDisplayOrientation().ty() get the y-component in meters in the world space frames,when I hold the phone,it is about 1.5 meters far to the ground(about the height of my arms to my feet),so if I change the y to frame.getCamera().getDisplayOrientedPose().ty() -15.8f,then the Anchor should be just on the ground.But the situation is the labels is still float in the air and has a few vertical distance to the ground.Then I change the numerial number much bigger than 1.5 such as 10,12,20,50 and so on,It is incredible that in some cases it is moves quickly until out of the range of view,in another cases,it has the almost the same vertical distance to the ground.In my scenario, the larger y is set, the higher it should be from the ground unless it is out of range,but it is not.

Is there any mistake in my understanding?How can i put the marker on the ground rather than in the air?Why can not y-component of the anchornode influence the height?
Thanks a lot for any help.

**Comment by Lilyxxzjzljy** [comment]
I have found the problem may be in the LocationNode.onUpdate,the node which is attached to the AnchorNode has been re-positioned by setWorldPosition,just as below:
`
            n.setWorldPosition(new Vector3(n.getWorldPosition().x, getHeight(), n.getWorldPosition().z));
            //n.setWorldPosition(new Vector3(n.getWorldPosition().x, -direction.y, n.getWorldPosition().z)); //here
         
            Quaternion lookRotation = Quaternion.lookRotation(direction, Vector3.up());
            n.setWorldRotation(lookRotation);
            n.setWorldScale(new Vector3(scale, scale, scale));
`
And I try to change the worldPosition,the AR label's height changes successfully.
